optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -151,6 +152,23 @@
     // join(..., planningVariableContainsCached(Talk::getPeriod, (Period a, Period b) -> a.overlaps(b)))
     // get the period value range, does a cartesian product on it, so it maps every period to an overlapping periodList
     // then keep an index from every period to all talks in an overlapping period (possible the same period)
+    public static <A, B, Property_ extends Comparable<Property_>> BiJoiner<A, B> overlapsOrMeets(
+            Function<A, Property_> leftStartMapping,
+            Function<A, Property_> leftEndMapping,
+            Function<B, Property_> rightStartMapping,
+            Function<B, Property_> rightEndMapping) {
+        return AbstractBiJoiner.merge(lessThanOrEqual(leftStartMapping, rightEndMapping),
```

Comment:
Considering that these are not simple Joiners anymore, I would:

- Extract their implementations into their own class.
- So that the tests for each also get their own class.
- And in that class, I'd make sure I cover all the boundary conditions in different test methods.

Reply:
But then how will people know these Joiners exists? People might already be used to all the Joiners being in the class `Joiners`. If we move it to a new class, we should at least mention it in the Javadocs of Joiners so they know there are other classes that contain Joiners (but not everyone read Javadocs; I suspect some people just use their IDE auto-complete which will hide the existence of these Joiners)

Reply:
Don'ŧ get me wrong. I think that the factory methods should stay in this class. But the implementations should be outside.